src/framework/theme/components/toggle/toggle.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { NbToggleComponent } from './toggle.component';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { Component, DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { ReactiveFormsModule, FormControl } from '@angular/forms';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { NbLayoutDirectionService } from '../../services/direction.service';
+import { NbIconModule } from '@nebular/theme';
+
+
+describe('Component: NbToggle', () => {
+  let toggle: NbToggleComponent;
+  let fixture: ComponentFixture<NbToggleComponent>;
+  let toggleInput: DebugElement;
+  let testContainerEl: HTMLElement;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [BrowserAnimationsModule, NbIconModule],
```

Comment:
I think it's better to import module of a component under the test instead of importing it's dependencies and redeclaring here. First, you ensure everything needed is exported by the component module. Second tests won't break you new dependency added to the component module.